core/src/main/scala/org/bitcoins/core/util/NumberUtil.scala

```diff
@@ -144,6 +144,49 @@ trait NumberUtil extends BitcoinSLogger {
     //should always be able to convert from uint5 => uint8
     u8sTry.get
   }
+
+  /** Expands the [[org.bitcoins.core.protocol.blockchain.BlockHeader.nBits nBits]]
+    * field given in a block header to the _actual_ target difficulty.
+    * See [[https://bitcoin.org/en/developer-reference#target-nbits developer reference]]
+    * for more information
+    *
+    * Meant to replicate this function in bitcoin core
+    * [[https://github.com/bitcoin/bitcoin/blob/2068f089c8b7b90eb4557d3f67ea0f0ed2059a23/src/arith_uint256.cpp#L206]]
+    * @param nBits
+    * @return
+    */
+  def targetExpansion(nBits: UInt32): BigInteger = {
+    //mantissa bytes without sign bit
+    val mantissaBytes = {
+      val withSignBit = nBits.bytes.takeRight(3)
+      val noSignBit = withSignBit.head & 0x7f
+      noSignBit.toByte +: withSignBit.tail
+    }
+
+    val significand = nBits.bytes.head
+
+    //if the most significant bit is set, we have a negative number
+    val signum = if ((0x80 & nBits.bytes(1)) != 0) {
```

Comment:
Is it cleaner to do interact with the bits directly rather than using bit masks? `if (nBits.bytes.bits(1)) ...`

I guess same applies for line 162